cloudbuild.yaml

```diff
@@ -23,29 +23,87 @@ steps:
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-windows-amd64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:windows_amd64
+      - //cmd/gke-exec-auth-plugin
+  # build gke-gcloud-auth-plugin binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-windows-amd64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:windows_amd64
+      - //cmd/gke-gcloud-auth-plugin
+  # build auth-provider-gcp binary
+  - name: 'gcr.io/cloud-builders/bazel'
+    args:
+      - --output_user_root=/workspace/bazel-root
+      - --output_base=/workspace/bazel-base-windows-amd64
+      - build
+      - --platforms=@io_bazel_rules_go//go/toolchain:windows_amd64
+      - //cmd/auth-provider-gcp
 # upload auth-provider-gcp. gke-gcloud-auth-plugin and gke-exec-auth-plugin binaries
 # TODO: figure out how to do this better, most probably getting rid of bazel
 artifacts:
   objects:
-    location: 'gs://k8s-staging-cloud-provider-gcp/auth-linux-amd64-$_PULL_BASE_REF'
+    location: 'gs://k8s-staging-cloud-provider-gcp/cloud-auth-provider-$_GIT_TAG'
     paths:
-      - '/workspace/plugin-binaries/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp'
-      - '/workspace/plugin-binaries/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-gcloud-auth-plugin/gke-gcloud-auth-plugin_/gke-gcloud-auth-plugin'
-      - '/workspace/plugin-binaries/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-exec-auth-plugin/gke-exec-auth-plugin_/gke-exec-auth-plugin'
+      - '/workspace/bazel-base-linux-amd64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp'
+      - '/workspace/bazel-base-linux-amd64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-gcloud-auth-plugin/gke-gcloud-auth-plugin_/gke-gcloud-auth-plugin'
+      - '/workspace/bazel-base-linux-amd64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-exec-auth-plugin/gke-exec-auth-plugin_/gke-exec-auth-plugin'
+      - '/workspace/bazel-base-linux-arm64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp'
+      - '/workspace/bazel-base-linux-arm64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-gcloud-auth-plugin/gke-gcloud-auth-plugin_/gke-gcloud-auth-plugin'
+      - '/workspace/bazel-base-linux-arm64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/gke-exec-auth-plugin/gke-exec-auth-plugin_/gke-exec-auth-plugin'
+      - '/workspace/bazel-base-windows-amd64/execroot/io_k8s_cloud_provider_gcp/bazel-out/k8-fastbuild/bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp'
```

Comment:
I think when building for windows binaries have `.exe` suffix added. At least that is what is happening with `auth-provider-gcp`.